### DISCUSSION TITLE
New Button Functionality: Configurable Time Between Clicks

### DISCUSSION THREAD

**Discussion by Jaimi5** [espressif/esp-iot-solution#345] [Ideas]
Hi! 

Hey everyone, and a special shoutout to @lijunru-hub! 

I’ve been diving into the button component recently and have developed a new feature that I would like to share with you all. It's an enhancement where you can now configure the time interval between clicks for detecting multiple clicks. 

Before I create a pull request, I wanted to reach out to the community, and especially @lijunru-hub. Your diagrams have been incredibly helpful, and they add a lot of clarity to the documentation. I was hoping we could discuss:

- The best way to introduce this feature through a pull request.
- If you could share some of your diagram magic to illustrate this new functionality. I’m keen on upgrading them to reflect this addition!

I'am looking forward to your insights and any suggestions you might have. Also, if anyone else has ideas or feedback, please don't hesitate to chime in. 

Thanks a lot!

**Comment by lijunru-hub** [comment]
This is a supported feature where you can call `iot_button_register_event_cb` to register multiple events and pass the `multiple_clicks_t` parameter to set the number of times for multiple clicks. However, the time interval between consecutive presses depends on the configuration inside the config.

Additionally, I think it would be very helpful to create a flowchart. I will add it as soon as possible.

**Reply by Jaimi5** [reply]
However, what I have looked the consecutive presses interval is defined by the SHORT_PRESS if I am not wrong. Which defines multiple clicks as clicking them "very fast". If you need to click 10 or 20 times it makes it a little difficult.

It would be great to create a flowchart to modify it in the future!

Sorry for the delay, I hadn't seen the message!

**Comment by lijunru-hub** [comment]
If your modification involves dynamically configuring the short press time for each button, could you describe your use case?

**Reply by Jaimi5** [reply]
I had added a new configuration that only involves the time between clicks when multiple clicks. This allows to define what consecutive clicks is defined. My use case is to allow the user click multiple times and have a time interval of a maximum of 3 seconds between clicks. A button is clicked, then a maximum of 3 seconds pass and you click again and again and it counts as multiple clicks.

**Reply by lijunru-hub** [reply]
Can you provide your code as an MR?

**Reply by lijunru-hub** [reply]
You can provide any time like 3 seconds when you create a button.
```
button_config_t btn_cfg = {
    .short_press_time = 3000,
};
```
Afterward, the interval for short press time will become 3000 ms. Is this the scenario you mentioned?
